game.py

```diff
@@ -0,0 +1,150 @@
+        elif self.level == 'medium':
+            self.characters = self.characters[:self.size - self.size // 3]
+        elif self.level == 'hard':
+            self.characters = self.characters[:self.size]
+
+    def set_values(self):
+        self.delete = set()
+        self.create_items = []
+
+    def play_game(self, row, column):
+        if self.winner:
+            return
+        self.set_values()
+        self.make_move(row, column)
+        if self.no_next_move():
+            return 'no more move'
+
+    def set_up_board(self):
+        for row in range(self.size):
+            for column in range(self.size):
+                character = random.choice(self.characters)
+                self.board[row][column] = character
+        if self.no_next_move():
+            self.set_up_board()
+
+    def make_move(self, row, column):
+        self.check_move(row, column)
+        if len(self.delete) < 3:
+            return
+        self.delete_items()
+        self.replace_items()
+        self.create_new_items()
+        if self.score >= self.target:
+            self.winner = True
+
+    def check_move(self, row, column):
+        current_item = self.board[row][column]
+        current_item_coordinates = row, column
+        neighbor_items = self.neighbor_items(row, column)
+        neighbor_coordinates = self.neighbor_coordinates(row, column)
+        for next_item, next_item_coordinates in zip(neighbor_items, neighbor_coordinates):
+            if current_item == next_item:
+                self.delete.add(current_item_coordinates)
+                if next_item_coordinates not in self.delete:
+                    self.check_move(*next_item_coordinates)
+
+    def no_next_move(self):
+        for row in range(self.size):
+            for column in range(self.size):
+                self.check_move(row, column)
+                equal_neighbors = self.delete
+                self.delete = set()
+                if len(equal_neighbors) >= 3:
+                    return False
+        return True
+
+    def neighbor_items(self, row, column):
+        left, right, lower, upper = None, None, None, None
+        if column > 0:
+            left = self.board[row][column - 1]
+        if column < self.size - 1:
+            right = self.board[row][column + 1]
+        if row < self.size - 1:
+            lower = self.board[row + 1][column]
+        if row > 0:
+            upper = self.board[row - 1][column]
+        return [left, right, lower, upper]
+
+    def neighbor_coordinates(self, row, column):
+        left, right, lower, upper = None, None, None, None
+        if column > 0:
+            left = row, column - 1
+        if column < self.size - 1:
+            right = row, column + 1
+        if row < self.size - 1:
+            lower = row + 1, column
+        if row > 0:
+            upper = row - 1, column
+        return [left, right, lower, upper]
+
+    @staticmethod
+    def row_above(row):
```

Comment:
what is this trying to check? naming is not clear

Reply:
this is checking if there is a row above the current row. I renamed it there_is_row_above